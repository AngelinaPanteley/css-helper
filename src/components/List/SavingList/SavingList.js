import React, { PureComponent } from 'react';

import styles from './SavingList.scss';
import SavingItem from './SavingItem/SavingItem';

class SavingList extends PureComponent {
  render() {
    return (
      <div>
        {this.props.items.map((item) => {
          return <SavingItem key={item.id} item={item} onEdit={this.props.onEdit} />
        })}
      </div>
    )
  }
}

export default SavingList;