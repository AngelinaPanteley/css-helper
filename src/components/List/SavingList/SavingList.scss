@import '../../../variables';

.Wrapper {
  max-height: calc(100vh - #{$header-height} - 160px);
  overflow: auto;
  border: 2px solid $color-border;
}