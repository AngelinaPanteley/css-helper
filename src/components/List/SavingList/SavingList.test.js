"use strict";

import React from 'react';
import renderer from 'react-test-renderer';
import { expect as chaiExpect } from 'chai';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16.1';
import { MemoryRouter } from "react-router-dom";
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import SavingList from './SavingList.js';
import reducer from '../../../store/reducers/index';

configure({ adapter: new Adapter() });

const store = createStore(reducer);

describe('>>>SavingList', () => {

  //SNAPSHOTS

  const props = {
    openItem: 'openItemId',
    items: [
      {
        id: '12345',
        title: 'title',
      }
    ],
    onClick: () => { },
    onEdit: () => { },
    onDelete: () => { },
  };

  const listComponent = <Provider store={store}><MemoryRouter>
    <SavingList {...props} />
  </MemoryRouter></Provider>;

  const component = renderer.create(listComponent);

  it('renders correctly', () => {
    let componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();
  });

  //SHALLOW RENDERING

  let wrapper;
  const shallowComponent = <SavingList {...props} />;

  beforeEach(() => {
    wrapper = shallow(shallowComponent);
  });

  it('renders SavingItems', () => {
    chaiExpect(wrapper.find('SavingItem')).to.have.lengthOf(1);
  });
});
