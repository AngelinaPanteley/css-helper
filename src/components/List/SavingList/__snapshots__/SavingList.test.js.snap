// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>SavingList renders correctly 1`] = `
<div>
  <div
    onClick={[Function]}
  >
    title
    <div>
      <button
        onClick={[Function]}
      >
        <i
          className="undefined undefined"
        />
      </button>
      <button
        onClick={[Function]}
      >
        <i
          className="undefined undefined"
        />
      </button>
    </div>
  </div>
  <div
    className=" "
  >
    <div>
      <div>
        <p>
          <span>
            Styles
          </span>
          <button
            onClick={[Function]}
          >
            <i
              className="undefined undefined"
            />
          </button>
        </p>
        <pre>
          <code>
            .undefined {
}
          </code>
        </pre>
      </div>
    </div>
    <div>
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "",
          }
        }
      />
    </div>
  </div>
</div>
`;
